packages/core/src/operator/listenForChallenges.ts

```diff
@@ -11,40 +9,45 @@ import { resilientEventListener } from "../utils/resilientEventListener.js";
  * @param callback - The callback function to be triggered when ChallengeSubmitted event is emitted.
  * @returns A function that can be called to stop listening for the event.
  */
-export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void): () => void {
-
-    // create a map to keep track of challengeNumbers that have called the callback
+export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void, onError?: (error: Error) => void): () => void {
     const challengeNumberMap: { [challengeNumber: string]: boolean } = {};
 
-    // listen for the ChallengeSubmitted event
     const listener = resilientEventListener({
         rpcUrl: config.arbitrumOneWebSocketUrl,
         contractAddress: config.refereeAddress,
         abi: RefereeAbi,
         eventName: "ChallengeSubmitted",
         log: console.info,
         callback: async (log, error) => {
-
-            if(error){
-                //TODO hanlde on error
+            if (error) {
+                console.error(error);
```

Comment:
This error log would be redundant with the already called log function in `resilientEventListener` on error and should be removed.

Reply:
Removed!